### ISSUE TITLE
Memory Leak in Image

### ISSUE THREAD

**Issue by hhelwich** [Automattic/node-canvas#1139]
There is a memory leak in the current stable version of node-canvas.
### Steps to Reproduce
1. Start some program to observe system memory
2.  Run the following code in node (set url to some Jpg file):
```js
const canvas = require('canvas');

const loadImageNTimes = n => {
    if (n > 0) {
        const image = new canvas.Image();
        image.onload = () => {
            // Uncomment the following line to free image memory:
            // image.src = '';
            loadImageNTimes(n - 1);
        };
        image.src = 'some.jpg'; // Set Url to an existing Jpg file
    }
};
loadImageNTimes(500);
```
3. You should see that the memory is now growing linearly
4. Uncomment the line `image.src = ''` and run the script again
5. The memory stays constant now

### Environment:
* node-canvas 1.6.10
* node 9.11.1 on Ubuntu 17.10

**Comment by Hakerh400** [comment]
Listener `image.onload` is called synchronously when `image.src` becomes a valid file path. Therefore, old image data structures cannot be deallocated until the last image finishes loading. You can fix it by adding `setTimeout` inside `onload` event listener.

**Comment by hhelwich** [comment]
Oh i didn't expect that :no_mouth: This solves the issue. Thank you!

**Comment by zbjornson** [comment]
This is also fixed by #1003 in 2.x.
